src/webui/api/torrentscontroller.cpp

```diff
@@ -527,15 +626,82 @@ void TorrentsController::addTrackersAction()
     const QString hash = params()["hash"];
 
     BitTorrent::TorrentHandle *const torrent = BitTorrent::Session::instance()->findTorrent(hash);
-    if (torrent) {
-        QList<BitTorrent::TrackerEntry> trackers;
-        foreach (QString url, params()["urls"].split('\n')) {
-            url = url.trimmed();
-            if (!url.isEmpty())
-                trackers << url;
+    if (!torrent)
+        throw APIError(APIErrorType::NotFound);
+
+    QList<BitTorrent::TrackerEntry> trackers;
+    for (QString urlStr : params()["urls"].split('\n')) {
+        urlStr = urlStr.trimmed();
+        if (QUrl(urlStr).isValid())
+            trackers << urlStr;
```

Comment:
The [QUrl docs](http://doc.qt.io/qt-5/qurl.html#QUrl-2) say that it fixes any mistakes on the string but you're passing the original string. Isn't it possible that the QUrl constructor will fix some problems that would make isValid() false?

Reply:
Good point, I'll use the QUrl string.